runtime/wasm/src/asc_abi/class.rs

```diff
@@ -87,7 +111,9 @@ impl<T: AscValue> AscType for ArrayBuffer<T> {
 #[repr(C)]
 pub(crate) struct TypedArray<T> {
     pub buffer: AscPtr<ArrayBuffer<T>>,
+    /// Byte position in `buffer` of the array start.
     byte_offset: u32,
+    /// Byte position in `buffer` of the array end.
     byte_length: u32,
```

Comment:
The field names here make it sound like the array end would be at `byte_offset + byte_length`. Are these comments right?

Reply:
The comment is correct, the field name is really misleading, but I chose to keep the same name [asc uses](https://github.com/AssemblyScript/assemblyscript/wiki/Memory-Layout-&-Management#arrays).

Reply:
Ah I see what you mean, that's weird. I'd vote for renaming the fields `byte_start` and `byte_end` or similar and leaving a comment noting the difference from the spec. Must be a mistake on the asc side.
(also: how about adding a comment mentioning that the 2nd ptr points to one byte past the end of the array?)

Reply:
I've renamed it, good point about it being one byte past, hadn't realized that.